src/SparkConduit.sol

```diff
@@ -132,31 +132,23 @@ contract SparkConduit is UpgradeableProxied, ISparkConduit, IInterestRateDataSou
     function withdraw(bytes32 ilk, address asset, uint256 maxAmount)
         external ilkAuth(ilk) returns (uint256 amount)
     {
-        uint256 liquidityAvailable
-            = IERC20(asset).balanceOf(IPool(pool).getReserveData(asset).aTokenAddress);
+        // Constrain the amount that can be withdrawn by the max amount
+        amount = _min(maxAmount, maxWithdraw(ilk, asset));
 
-        // Constrain by the amount of liquidity available of the token
-        amount = liquidityAvailable < maxAmount ? liquidityAvailable : maxAmount;
-
-        // Constrain by the amount of deposits this ilk has
-        uint256 ilkDeposits = shares[asset][ilk].rayMul(IPool(pool).getReserveNormalizedIncome(asset));
-        amount = ilkDeposits < amount ? ilkDeposits : amount;
-
-        uint256 removedShares = amount.rayDiv(IPool(pool).getReserveNormalizedIncome(asset));
+        uint256 removedShares = _convertToShares(asset, amount);
 
+        // Reduce share accounting by the amount withdrawn
         shares[asset][ilk] -= removedShares;
         totalShares[asset] -= removedShares;
 
         uint256 withdrawals = requestedShares[asset][ilk];
```

Comment:
This variable should be renamed to `requestedShares_` or something to reduce confusion.

Reply:
Fixed, updated surrounding variable names as well